src/conn/conn_log.c

```diff
@@ -447,7 +447,14 @@ __log_prealloc_once(WT_SESSION_IMPL *session)
 		__wt_verbose(session, WT_VERB_LOG,
 		    "Missed %" PRIu32 ". Now pre-allocating up to %" PRIu32,
 		    log->prep_missed, conn->log_prealloc);
-	}
+	} else if (reccount > conn->log_prealloc / 2)
+		/*
+		 * If we used less than half, then adjust how many we allocate
+		 * downward. Cut in half, plus 1 so that we always allocate
+		 * at least one.
+		 */
+		conn->log_prealloc = conn->log_prealloc / 2 + 1;
```

Comment:
I noticed that at the time of checkpoint not as many PrepLog files would be renamed to log files as usual. The log server thread here would still think that so many prep logs are not required and adjust downward. Would that be right?

Also should the downward adjustment be gradual e.g. reducing one file at a time. Sometimes `reccount` might show just a temporary reduction in use of Prep logs. In that situation the `log_prealloc` might need to be re-adjusted upwards.